### PR TITLE
ADD lib.py to make argnorm usable as a library

### PR DESCRIPTION
**Pull request by Vedanth-Ramji**
Making argNorm better suited as a library. 

- Providing a separate `map_to_aro` function so argNorm can be used without the `normalizers`
- Separated `get_aro_mapping_table` function from `BaseNormalizer` so it can be used independently

### CODE REVIEW

File: argnorm/normalizers.py [hunk 1]
```diff
@@ -221,4 +187,4 @@ def preprocess_ref_genes(self, ref_genes):
             megares=lambda x: x.split('|')[0],
             argannot=lambda x: x.split('~~~')[-1]
         )
-        return ref_genes.apply(process_funcs_by_db[self.database])
+        return ref_genes.apply(process_funcs_by_db[self.database])
```
Comment (luispedro) [inline argnorm/normalizers.py:190]: Is this needed?

File: argnorm/general.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+def map_to_aro(gene, database):
+    if database not in ['ncbi', 'deeparg', 'resfinder', 'sarg', 'megares', 'argannot']:
+        raise Exception(f'{database} is not a supported database.')
+    
+    mapping_table = get_aro_mapping_table(database).set_index('Original ID')
+
+    try:
+        result = mapping_table.loc[gene, 'ARO']
+    except:
```
Comment (luispedro) [inline argnorm/general.py:44]: Naked `except` is not good

**Review by luispedro — CHANGES_REQUESTED** [review]
I'm not sure about splitting to `general`. Is the goal that people import it as 

```python
from argnorm import general
```

?

Also, tests are needed

**Comment by Vedanth-Ramji** [comment]
Yes, well now that I have renamed it to 'lib' it should be something like: `from argnorm import lib` or `from argnorm.lib import *`. 

Is that ok?

File: argnorm/normalizers.py [hunk 3]
```diff
@@ -1,31 +1,12 @@
 import os
 import pandas as pd
 from .drug_categorization import confers_resistance_to, drugs_to_drug_classes
-
-ORIGINAL_ID_COL = 'Original ID'
-MAPPING_TABLE_ARO_COL = 'ARO'
-TARGET_ARO_COL = 'ARO'
+from .lib import *
```
Comment (luispedro) [inline argnorm/normalizers.py:4]: Please avoid the `import *`

**Review by luispedro — CHANGES_REQUESTED** [review]

**Comment by luispedro** [comment]
Can you some notes to the Changelog about this?

**Review by luispedro — APPROVED** [review]

**Pull request merged**